torchvision/models/_api.py

```diff
@@ -140,3 +143,54 @@ def _get_enum_from_fn(fn: Callable) -> WeightsEnum:
         )
 
     return cast(WeightsEnum, weights_enum)
+
+
+M = TypeVar("M", bound=Type[nn.Module])
+
+BUILTIN_MODELS = {}
+
+
+def register_model(name: Optional[str] = None, overwrite: bool = False) -> Callable[[Callable[..., M]], Callable[..., M]]:
```

Comment:
Q: what do we need the `overwrite` parameter for? IIUC the registration decorator is private, so this must come  from an internal need?

Reply:
ClassyVision's registration mechanism had it as an option, so I thought to do the same here. You are right to say that internally, we can fully control this and won't need the overwrite feature. I suspect that eventually we might need it for fb internal bits but if you prefer it we can drop it for now.